csp_billing_adapter/adapter.py

```diff
@@ -147,6 +150,11 @@ def main() -> None:
     except SystemExit as e:
         # user exception, program aborted by user
         sys.exit(e)
+    except CSPBillingAdapterException as e:
+        # csp_billing_adapter specific exception
+        log.error('CSP Billing Adapter error: {0}'.format(e))
+        traceback.print_exc()
+        sys.exit(2)
```

Comment:
i don't think the adapter should exit. Otherwise it would stop getting usage data. I think it should log error, update the csp-config with error message and failed state, then exit current loop.

Reply:
Since it's in a container the container would go into a infinite crash loop until removed which I don't think is desired.

Reply:
Okeydokey...

Just want to confirm what you mean by "exit the current loop"?

If we update the csp-config with the error information and to indicate a failure, but do not update the cached next_bill_time or remove the problem usage records, and exit the event_loop_handler() then we the next time we get woken up we will repeat the same steps and hit the same issue again... So forever after, until an updated config gets introduced, we will repeat that pattern...

If that what you intend?

Reply:
Yeah, this issue is a deployment time issue. Thus it "shouldn't" happen in production. If it does I think we want the adapters to continue caching usage. Otherwise usage is lost forever.

Reply:
Ok that is what I'm working on implementing.

So if they do keep running, logging usage data, but also saving the error in the csp_config, if we published a new container image with a fixed config embedded in it, and the adapter app was restarted in k8s to pick up that new container image, the problem could potentially resolve itself, IIUC?

Reply:
Should be addressed in my latest changes.